data/data/azure/vnet/public-lb.tf

```diff
@@ -12,49 +14,59 @@ resource "azurerm_public_ip" "cluster_public_ip" {
 }
 
 data "azurerm_public_ip" "cluster_public_ip" {
-  name                = azurerm_public_ip.cluster_public_ip.name
+  count = var.private ? 0 : 1
+
+  name                = azurerm_public_ip.cluster_public_ip[0].name
   resource_group_name = var.resource_group_name
 }
 
 resource "azurerm_lb" "public" {
+  count = var.private ? 0 : 1
+
   sku                 = "Standard"
   name                = "${var.cluster_id}-public-lb"
   resource_group_name = var.resource_group_name
   location            = var.region
 
   frontend_ip_configuration {
     name                 = local.public_lb_frontend_ip_configuration_name
-    public_ip_address_id = azurerm_public_ip.cluster_public_ip.id
+    public_ip_address_id = azurerm_public_ip.cluster_public_ip[0].id
   }
 }
 
 resource "azurerm_lb_backend_address_pool" "master_public_lb_pool" {
+  count = var.private ? 0 : 1
+
   resource_group_name = var.resource_group_name
-  loadbalancer_id     = azurerm_lb.public.id
+  loadbalancer_id     = azurerm_lb.public[0].id
   name                = "${var.cluster_id}-public-lb-control-plane"
 }
 
 resource "azurerm_lb_rule" "public_lb_rule_api_internal" {
+  count = var.private ? 0 : 1
```

Comment:
this seems to be incorrect, private lb is required in private only mode, maybe drop this count here

Reply:
ahh, nvm